main.tf

```diff
@@ -21,6 +21,16 @@ resource "aws_elasticache_replication_group" "redis" {
   maintenance_window            = var.redis_maintenance_window
   snapshot_window               = var.redis_snapshot_window
   snapshot_retention_limit      = var.redis_snapshot_retention_limit
+  availability_zones            = var.availability_zones
+  
+  dynamic "cluster_mode" {
+    for_each = var.redis_clusters == null ? [var.redis_clusters] : []
```

Comment:
This looks wrong. It says: 
if redis_cluster is null "then" [null] else []
I would advice to try to plan using your branch and see where the error is.
What you probably want is:
if something is true then do this once otherwise []